neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/impl/BaseNeo4jSchemaDefiner.java

```diff
@@ -24,47 +28,80 @@
 import org.hibernate.ogm.datastore.neo4j.logging.impl.Log;
 import org.hibernate.ogm.datastore.neo4j.logging.impl.LoggerFactory;
 import org.hibernate.ogm.datastore.spi.BaseSchemaDefiner;
+import org.hibernate.ogm.datastore.spi.DatastoreProvider;
+import org.hibernate.ogm.model.key.spi.IdSourceKeyMetadata;
+import org.hibernate.service.spi.ServiceRegistryImplementor;
+import org.hibernate.tool.hbm2ddl.UniqueConstraintSchemaUpdateStrategy;
 import org.jboss.logging.Logger.Level;
 import org.neo4j.graphdb.Label;
+
 /**
+ * Allows the reuse of the logic for the creation of unique constraints when the schema is created.
+ *
  * @author Davide D'Alto
  */
-public abstract class BaseNeo4jSchemaDefiner<T> extends BaseSchemaDefiner {
+public abstract class BaseNeo4jSchemaDefiner extends BaseSchemaDefiner {
+
+	public static class UniqueConstraintDetails {
+		private final Label label;
+		private final String property;
+
+		public UniqueConstraintDetails(Label label, String property) {
+			super();
+			this.label = label;
+			this.property = property;
+		}
+
+		public Label getLabel() {
+			return label;
+		}
+
+		public String getProperty() {
+			return property;
+		}
+
+		@Override
+		public String toString() {
+			return "UniqueConstraintDetails [label=" + label + ", property=" + property + "]";
+		}
+	}
 
 	private static final Log log = LoggerFactory.getLogger();
 
-	protected void addUniqueConstraints(T neo4jDb, Database database) {
+	protected void createUniqueConstraints(DatastoreProvider provider, Database database) {
+		List<UniqueConstraintDetails> constraints = new ArrayList<>();
 		for ( Namespace namespace : database.getNamespaces() ) {
 			for ( Table table : namespace.getTables() ) {
 				if ( table.isPhysicalTable() ) {
 					Label label = label( table.getName() );
 					PrimaryKey primaryKey = table.getPrimaryKey();
-					createConstraint( neo4jDb, table, label, primaryKey );
+					createConstraint( constraints, table, label, primaryKey );
 					@SuppressWarnings("unchecked")
 					Iterator<Column> columnIterator = table.getColumnIterator();
 					while ( columnIterator.hasNext() ) {
 						Column column = columnIterator.next();
 						if ( column.isUnique() ) {
-							createUniqueConstraintIfMissing( neo4jDb, label, column.getName() );
+							constraints.add( new UniqueConstraintDetails( label, column.getName() ) );
 						}
 					}
 					Iterator<UniqueKey> uniqueKeyIterator = table.getUniqueKeyIterator();
 					while ( uniqueKeyIterator.hasNext() ) {
-						createConstraint( neo4jDb, table, label, uniqueKeyIterator.next() );
+						createConstraint( constraints, table, label, uniqueKeyIterator.next() );
```

Comment:
Maybe rename this `addConstraint` as it does not create the constraints anymore.